src/filesystem.h

```diff
@@ -29,7 +29,7 @@
 namespace sentencepiece {
 namespace filesystem {
 
-typedef std::variant<absl::string_view, std::string> ps_string;
+typedef std::variant<absl::string_view, std::shared_ptr<std::string>> ps_string;
```

Comment:
```suggestion
using ps_string = std::variant<absl::string_view, std::shared_ptr<std::string>>;
```

`typedef` is entirely superseded by `using` (i.e everything `typedef` can do, `using` does, and more).